packages/ui5-task-flatten-library/lib/flatten-library.js

```diff
@@ -0,0 +1,52 @@
+const log = require("@ui5/logger").getLogger("builder:customtask:ui5-task-flatten-library");
+
+/**
+ * Task to flatten the library folder structure.
+ * This is required for deployments to SAP NetWeaver.
+ *
+ * @param {Object} parameters Parameters
+ * @param {DuplexCollection} parameters.workspace DuplexCollection to read and write files
+ * @param {module:@ui5/builder.tasks.TaskUtil} [parameters.taskUtil] TaskUtil
+ * @param {Object} parameters.options Options
+ * @param {Object} parameters.options.projectName Project name
+ * @param {Object} parameters.options.projectNamespace Project namespace
+ * @returns {Promise<undefined>} Promise resolving with undefined once data has been written
+ */
+module.exports = async function ({ workspace, taskUtil, options }) {
+
+  if (!taskUtil.isRootProject()) {
+    // Flattening is only done when library is built as root project
+    log.info(`Skipping execution. Current project '${options.projectName}' is not the root project.`);
+    return;
+  }
+
+  if (!options.projectNamespace) {
```

Comment:
please put the documentation for this config-time param into the `README` and/or explain its' usage in the extesion

Reply:
That's a default parameter that every task receives: https://sap.github.io/ui5-tooling/pages/extensibility/CustomTasks/#task-implementation